### PR TITLE
Improve provider update check error message

### PR DESCRIPTION
**Pull request by irmatov**
Rework unit tests to not require network access

### CODE REVIEW

File: pkg/ui/console/client.go [hunk 1]
```diff
@@ -92,7 +93,7 @@ func (c Client) DownloadProviders(ctx context.Context) error {
 		c.updater.Wait()
 	}
 	ui.ColorizedOutput(ui.ColorProgress, "Finished provider initialization...\n\n")
-	updates, _ := c.c.CheckForProviderUpdates(ctx)
+	updates := client.CheckForProviderUpdates(ctx, registry.GetLatestRelease, c.c.Providers, c.c.Logger)
```
Comment (yevgenypats) [inline pkg/ui/console/client.go:96]: why migrate it from method level and then pass everything as parameters?

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Still when running with latest and --enable-console-log I see a pretty confusing message:

```
2:28PM INF Finished provider initialization...
2:28PM WRN Update check failed error="bad local version: provider aws, version latest" provider=aws
```

**Comment by irmatov** [comment]
Now it's clear what's the problem..

File: pkg/ui/console/client.go (hunk 1 above)
Comment (irmatov) [reply pkg/ui/console/client.go:96]: It makes it more of a pure function, easier to test and easier to reason about. Doesn't matter much in this case, I reverted it.

**Comment by irmatov** [comment]
Should be fixed now.

**Comment by yevgenypats** [comment]
@irmatov wdyt about this approach - https://github.com/cloudquery/cloudquery/pull/388 ?

**Comment by irmatov** [comment]
@yevgenypats it is mostly the same. Here I rework unit tests a bit to not depend on external repo (and network access, and even the DB availability) just to check how our update detection works.

**Review by spangenberg — APPROVED** [review]
LGTM

**Comment by yevgenypats** [comment]
Fixed by https://github.com/cloudquery/cloudquery/pull/391 . Per the tests I think if the tests are simple and short it's better to focus on integration e2e tests.

**Pull request closed without merging**